src/main/scala/com/exini/sbx/dicom/streams/DicomStreamOps.scala

```diff
@@ -394,16 +395,34 @@ trait DicomStreamOps {
     if (tagFilterFlow.isDefined || reverseAnonymization) {
       val label = "collect-reverse"
 
-      groupLengthDiscardFilter
-        .via(toIndeterminateLengthSequences)
-        .via(tagFilterFlow.getOrElse(identityFlow))
+      def filterFlow =
+        if (tagFilterFlow.isDefined) // filtering will happen, must assure no explicit length element groups
```

Comment:
Here is the bugfix: If filtering is performed we must make sure dicom stream has no explicit length attribute groups.